analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyCollectionsShouldNotBeEnumerated.vb

```diff
@@ -193,7 +194,47 @@ Public Class CollectionTests
 
     Public Sub Casing()
         Dim List As New List(Of Integer)
-        List.cLEAR()                    ' Noncompliant
+        List.Clear()                    ' Noncompliant
+    End Sub
+
+    Public Sub LearnConditions_Size(Condition As Boolean)
+        Dim IsNull As List(Of Integer) = Nothing
+        Dim Empty As New List(Of Integer)
+        Dim NotEmpty As New List(Of Integer) From {1, 2, 3}
+
+        If If(IsNull, Empty).Count = 0 Then Empty.Clear()  ' Noncompliant
+        If If(IsNull, NotEmpty).Count = 0 Then Empty.Clear()  ' Compliant, unreachable
+        If If(Condition, Empty, Empty).Count = 0 Then
+            Empty.Clear()  ' Noncompliant
+        Else
+            Empty.Clear()  ' Compliant, unreachable
+        End If
+        If Empty.Count = 0 Then
+            Empty.Clear()  ' Noncompliant
+        Else
+            Empty.Clear()  ' Compliant, unreachable
+        End If
+        If Empty.Count() = 0 Then
+            Empty.Clear()  ' Noncompliant
+        Else
+            Empty.Clear()  ' Compliant, unreachable
+        End If
+        If Enumerable.Count(Empty) = 0 Then
+            Empty.Clear()  ' Noncompliant
+        Else
+            Empty.Clear()  ' Compliant, unreachable
+        End If
+        NotEmpty.Clear()   ' Compliant, prevents LVA from throwing notEmpty away during reference capture
+    End Sub
+
+    Public Sub Enumerable_ExtensionIsInstance()
+        Dim Empty As New List(Of Integer)
+        Dim AsIEnumerable As IEnumerable(Of Integer) = Empty
+        If AsIEnumerable.Count = 0 Then ' This is invoked as instance, not as a static call
```

Comment:
Educational: This a pure VB thing, right? Why is it calling this as instance?

Reply:
Yes, it's VB-only. 
There are some Roslyn internals about how they transform `IMethodSymbols` in a weird way sometimes. Not sure why exactly.

At least this one is understandable. For example [this thing](https://github.com/SonarSource/sonar-dotnet/blob/ba7978e771d0df3c83d841872821997b2702d392/analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.vb#L179) is really weird.